include/emp/web/Color.hpp

```diff
@@ -0,0 +1,366 @@
+        i = ::round(i);  // Seems to be what Chrome does (vs truncation).
+        return i < 0 ? 0 : i > 255 ? 255 : uint8_t(i);
+    }
+
+    template <typename T>
+    float clamp_css_float(T f) {  // Clamp to float 0.0 .. 1.0.
+        return f < 0 ? 0 : f > 1 ? 1 : float(f);
+    }
+
+    float parseFloat(const std::string& str) {
+        return strtof(str.c_str(), nullptr);
+    }
+
+    int64_t parseInt(const std::string& str, uint8_t base = 10) {
+        return strtoll(str.c_str(), nullptr, base);
+    }
+
+    uint8_t parse_css_int(const std::string& str) {  // int or percentage.
+        if (str.length() && str.back() == '%') {
+            return clamp_css_byte(parseFloat(str) / 100.0f * 255.0f);
+        } else {
+            return clamp_css_byte(parseInt(str));
+        }
+    }
+
+    float parse_css_float(const std::string& str) {  // float or percentage.
+        if (str.length() && str.back() == '%') {
+            return clamp_css_float(parseFloat(str) / 100.0f);
+        } else {
+            return clamp_css_float(parseFloat(str));
+        }
+    }
+
+    float css_hue_to_rgb(float m1, float m2, float h) {
+        if (h < 0.0f) {
+            h += 1.0f;
+        } else if (h > 1.0f) {
+            h -= 1.0f;
+        }
+
+        if (h * 6.0f < 1.0f) {
+            return m1 + (m2 - m1) * h * 6.0f;
+        }
+        if (h * 2.0f < 1.0f) {
+            return m2;
+        }
+        if (h * 3.0f < 2.0f) {
+            return m1 + (m2 - m1) * (2.0f / 3.0f - h) * 6.0f;
+        }
+        return m1;
+    }
+
+    emp::vector<std::string> split(const std::string& s, char delim) {
+        emp::vector<std::string> elems;
+        std::stringstream ss(s);
+        std::string item;
+        while (std::getline(ss, item, delim)) {
+            elems.push_back(item);
+        }
+        return elems;
+    }
+
+  }
+
+  Color::Color( const std::string& css_str ) {
+   std::string str = css_str;
+
+    // Remove all whitespace, not compliant, but should just be more accepting.
+    str.erase(std::remove(str.begin(), str.end(), ' '), str.end());
+
+    // Convert to lowercase.
+    std::transform(str.begin(), str.end(), str.begin(), ::tolower);
+
+    for (const auto& namedColor : color_impl::namedColors) {
+        if (str == namedColor.name) {
+            *this = namedColor.color;
+            return;
+        }
+    }
+
+    ParseColor(str, css_str);
+    
+    return;
+
+  }
+  
+  // Handle all color detection and parsing from the string
+  Color & Color::ParseColor(const std::string str, const std::string& css_str) {
+      // #abc and #abc123 syntax.
+    if (DetectABC(str))
+        return ParseABC(str, css_str);
+
+    size_t op = str.find_first_of('('), ep = str.find_first_of(')');
+    if (op != std::string::npos && ep + 1 == str.length()) {
+        const std::string fname = str.substr(0, op);
+        const std::vector<std::string> params = color_impl::split(str.substr(op + 1, ep - (op + 1)), ',');
```

Comment:
Can we name this something more descriptive than params? We might call them tokens or words, maybe with some sort of adjective